log/src/main/java/kcl/teamIndexZero/traffic/log/FileIO/FileOutput.java

```diff
@@ -83,52 +89,80 @@ public boolean closeWriter() {
 
     /**
      * Writes a string to the end of the file
+     *
      * @param string String to append to file
-     * @exception IOException when the file cannot be accessed
+     * @throws IOException when the file cannot be accessed
      */
-    public void appendString( String string ) throws InvalidPathException, IOException {
+    public synchronized void appendString(String string) throws InvalidPathException, IOException {
```

Comment:
Correct - for now that's probably irrelevant. 

If you mean you want to retain an order of messages though they are coming from multiple threads - that's probably about impossible, as thread schedule is from our point of view indeterministic and controlled from outside (by OS). So, say we won't make sure that message A goes in before message B. 

What we can make sure with queue is that messages will not overlap - i.e. there may be situations otherwise when you have a message 1 starts, then message 2 breaks it right int the log - and we get a nonsense. So that they are atomic. 


But you are right - let's revise this again later.